Completed Lesson-6

borisshavrin opened the pull request.

borisshavrin committed:
Completed Lesson-6

borisshavrin committed:
Дополнил файлы

borisshavrin commented on task2.py:
```diff
@@ -0,0 +1,23 @@
+    всего дорожного полотна. Использовать формулу: длина * ширина * масса асфальта для покрытия
+    одного кв метра дороги асфальтом, толщиной в 1 см * число см толщины полотна. Проверить работу метода.
+Например: 20м * 5000м * 25кг * 5см = 12500 т"""
+
```
# Или нужно было так передать аргументы объекта?
class Road: 
    _lenght = 0
    _widht = 0
    
    def total_weight(self): 
        total = self._widht * self._lenght * 25 * 5 
        print(f'Масса асфальта, необходимого для покрытия всего дорожного полотна: {total/1000} т.') 


r = Road() 
r._lenght = 5000   # аргумент длины объекта
r._widht = 20        # аргумент ширины объекта
r.total_weight() 

print(Road._lenght) 
print(Road._widht)
print(r._widht)
print(r._lenght)

pantonov-gb commented on task1.py:
```diff
@@ -0,0 +1,26 @@
+        time.sleep(2)
+        print((self.__color[2]))
+        time.sleep(2)
+        s
```
а это кто тут?

pantonov-gb commented on task2.py:
```diff
@@ -0,0 +1,23 @@
+
+    def total_weight(self, lenght, widht):
+        Road._lenght = lenght
+        Road._widht = widht
```
Тут логичнее было бы в конструкторе устанавливать эти атрибуты у экземпляра, а не у класса, т е `self._length = length`

pantonov-gb commented on task3.py:
```diff
@@ -0,0 +1,42 @@
+class Position(Worker):                             # Класс на базе Worker
+
+    def get_full_name(self, name, surname):         # метод получения полного имени сотрудника
+        Worker.name = name
```
self.name

pantonov-gb reviewed (commented):
Надо доработать

borisshavrin committed:
Update task1.py

Удалил случайный символ s

borisshavrin committed:
Update task2.py

передача аргументов в методе 
self._lenght = lenght
self._widht = widht

borisshavrin committed:
Update task3.py

borisshavrin committed:
Update task2

borisshavrin committed:
update task2

borisshavrin committed:
update task3

borisshavrin committed:
Update task4

Добавлен конструктор, изменены переопределения методов, добавлено условие для перехода между методами go и stop

borisshavrin committed:
Update task5

Добавлен конструктор и исправлено переопределение метода, изменен вывод

pantonov-gb reviewed (approved):
Ок
